osv/third_party/univers/alpine.py

```diff
@@ -146,19 +146,26 @@ def __ge__(self, other):
 @attr.s(frozen=True, order=False, eq=False, hash=True)
 class AlpineLinuxVersion(Version):
   """Alpine linux version"""
-  # E.g. For this version (1.9.5p2-r3), the following regex
-  # extracts (1.9.5p2) and (3)
+  # E.g. For this version (1.9.5_p2-r3), the following regex
+  # extracts (1.9.5_p2) and (3)
   version_extractor = re.compile(r'(.+?)(?:-r(\d+))?$')
   
+  # Some suffixes are not separated with an underscore. E.g. 1.9.5p2
+  # This should find them for inserting an underscore (replace with r'\1_\2')
+  patch_finder = re.compile(r'(\d+)(p\d+)')
+
   @classmethod
   def build_value(cls, string: str):
     search = cls.version_extractor.search(string)
-    return (search.group(1), int(search.group(2) or 0))
+    main_group = search.group(1)
+    main_group_patched = re.sub(cls.patch_finder, r'\1_\2', main_group, 1)
```

Comment:
nit: instead of `patch_finder`,  should we just make a classmethod (`add_underscores`) that can be called in both places here, instead of duplicating the re.sub call?